### PR TITLE
Updated the holochainClient to make one WebSocket Connection

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

**Review by JettTech — APPROVED** [review]
LGTM. Thanks for updating this. :tada:  

Also, happy to leave the error catch to 10 tries/call for now, but let's talk about this and see if there is an optimal number.

File: src/holochainClient.js [hunk 1]
```diff
@@ -131,14 +133,28 @@ async function initAndGetHolochainClient () {
...
 }
+async function initAndGetHolochainClient () {
+  let counter = 0
+  while (hcFlag) {
+    counter++
+    await wait(100)
+    if (counter === 10) {
+      hcFlag = false
+    }
+  }
```
Comment (robbiecarlton) [inline src/holochainClient.js]: isn't this equivalent to
```
if (hcFlag) {
  await wait(1000)
  hcFlag = false
}
```
?

File: src/holochainClient.js [hunk 2]
```diff
@@ -111,9 +112,10 @@ export function conductorInstanceIdbyDnaAlias (instanceId) {
 }
 
 let holochainClient
+let hcFlag = false
 
-async function initAndGetHolochainClient () {
-  if (holochainClient) return holochainClient
+async function init () {
```
Comment (robbiecarlton) [inline src/holochainClient.js]: on the fence about this, but I think I'd prefer this be called `initHolochainClient`.

It makes the specific difference between the two functions clearer.

File: src/holochainClient.js [hunk 3]
```diff
@@ -111,9 +112,10 @@ export function conductorInstanceIdbyDnaAlias (instanceId) {
 }
 
 let holochainClient
+let hcFlag = false
```
Comment (robbiecarlton) [inline src/holochainClient.js]: can we call this something more descriptive?

`isInitiatingHcConnection` or similar

File: package.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
     "test:lint": "standard",
     "test:unit": "node scripts/test.js --config=./jest.unit.config.js",
     "test": "npm run test:lint && npm run test:unit",
-    "test:ci": "CI=1 npm run test",
+    "test:ci": "CI=1 npm run test:unit",
```
Comment (robbiecarlton) [inline package.json:24]: 😆 I feel like we make this change all the time and somehow it never sticks

**Review by robbiecarlton — CHANGES_REQUESTED** [review]

File: src/holochainClient.js (hunk 1 above)
Comment (robbiecarlton) [reply src/holochainClient.js]: oh, I get it.

Going to think about if there's a more readable way to do this.

File: src/holochainClient.js [hunk 5]
```diff
@@ -131,14 +133,28 @@ async function initAndGetHolochainClient () {
...
     if (HOLOCHAIN_LOGGING) {
       console.log('😞 Holochain client connection failed -- ', error.toString())
     }
+    isInitiatingHcConnection = false
     throw (error)
   }
 }
+async function initAndGetHolochainClient () {
+  let counter = 0
+  while (isInitiatingHcConnection) {
```
Comment (robbiecarlton) [inline src/holochainClient.js:154]: can we add a comment calling out that this variable is being changed in a different function running in parallel?

**Review by robbiecarlton — CHANGES_REQUESTED** [review]
One more change request, then it's good to merge 👍

**Review by robbiecarlton — APPROVED** [review]

**Pull request merged**